contracts/cw3-fixed-multisig/src/contract.rs

```diff
@@ -49,7 +50,8 @@ pub fn instantiate(
     // add all voters
     for voter in msg.voters.iter() {
         let key = deps.api.addr_validate(&voter.addr)?;
-        VOTERS.save(deps.storage, &key, &voter.weight)?;
+        VOTERS.insert(deps.storage, &key, &voter.weight)?;
+        VOTER_ADDRESSES.insert(deps.storage, &key)?;
```

Comment:
what about using `.iter_keys()`? https://github.com/scrtlabs/secret-toolkit/blob/master/packages/storage/src/keymap.rs#L560-L565

my concern is the keys from the VOTERS map going out of sync from the VOTER_ADDRESSES list, would be easier to manage one source of truth instead

Reply:
That works but then we're back to performing potentially prohibitively expensive sorting operations as the number of voters increases. The idea behind the BST was to allow the CW3 to scale to support an arbitrary amount of voters without significant increases in computational cost. But if that's not an intended use case I guess it doesn't matter. In that case the BST is pretty useless too.

I don't see how they are going to get out of sync as they are always mutated together; the only scenario I see is if `VOTERS` is successfully mutated and `VOTER_ADDRESSES` fails with an error, but I now realize this was probably the motivation behind the other snippet you commented on:
```rust
for addr in addresses {
    let weight = match VOTERS.get(deps.storage, &addr) {
        Some(weight) => weight,
        None => continue,
    };
    ...
```
Although we probably need to swap the insertion order so that there are always at worst more addresses than voters:
```rust
VOTER_ADDRESSES.insert(deps.storage, &key)?;
VOTERS.insert(deps.storage, &key, &voter.weight)?;
```